compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp

```diff
@@ -104,21 +104,25 @@ struct FuseElementwiseOpsWithMultipleUses
     if (!consumerMarker) return failure();
 
     auto fusedOperandIt =
-        llvm::find_if(consumerOp->getOpOperands(), [](OpOperand &operand) {
+        llvm::find_if(consumerOp->getOpOperands(), [&](OpOperand &operand) {
           Operation *operandProducer = operand.get().getDefiningOp();
           if (!operandProducer) return false;
           auto producerMarker = operandProducer->getAttrOfType<IntegerAttr>(
               getProducerAttributeName());
           if (!producerMarker) return false;
-          return true;
+          return consumerMarker.getValue() == producerMarker.getValue();
```

Comment:
Its nice to check, but did you run into an issue with it? This should have never happened by construction.